tests/containers/validate_btrfs.pm

```diff
@@ -0,0 +1,79 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test Docker’s btrfs storage driver features for image and container management
+# Among these features are block-level operations, thin provisioning, copy-on-write snapshots,
+# and ease of administration. You can easily combine multiple physical block devices into a single Btrfs filesystem.
+# The scenario illustrates the build of a sle image and then reuse the same Dockerfile to check that
+# the thin partitioning are used. Then we check the block subvolumes.
+# Finally we test the disk administration filling up the subvolume mounted for the images
+# and try to pull another image. With btrfs we should be able to add another disk and be able to
+# continue when the docker partition is fulled up.
+# Maintainer: qac team <qa-c@suse.de>
+
+use Mojo::Base qw(consoletest);
+use testapi;
+use containers::runtime;
+
+sub _sanity_test_btrfs {
+    my ($rt, $dev_path) = @_;
+    my $dockerfile_path = '/root/sle_base_image/docker_build';
+    my $btrfs_head      = '/tmp/subvolumes_saved';
+    $rt->info(property => 'Driver', value => 'btrfs');
+    $rt->build($dockerfile_path, 'huge_image');
+    assert_script_run "btrfs fi df $dev_path/btrfs/";
+    assert_script_run "ls -td $dev_path/btrfs/subvolumes/* | head -n 1 > $btrfs_head";
+    validate_script_output "df -h |grep var", sub { m/\/dev\/vda.+[1-6]\d?%/ };
+    $rt->up('huge_image', cmd => 'fallocate -l 8.7G bigfile.stty');
+    # partition should be full
+    validate_script_output "df -h |grep var",       sub { m/\/dev\/vda.+\s+(9[7-9]|100)%/ };
+    validate_script_output "btrfs fi df $dev_path", sub { m/^Data.+total=8.*GiB, used=8.*GiB/ };
+}
+
+sub _test_btrfs_balancing {
+    my ($dev_path) = shift;
+    assert_script_run qq(btrfs balance start $dev_path), timeout => 600;
```

Comment:
`btrfs balance start /devicepath` usually prints the following warning, including a wait time of 10 seconds:

    WARNING:
    
            Full balance without filters requested. This operation is very
            intense and takes potentially very long. It is recommended to
            use the balance filters to narrow down the scope of balance.
            Use 'btrfs balance start --full-balance' option to skip this
            warning. The operation will start in 10 seconds.
            Use Ctrl-C to stop it.
    10 9 8 7 6 5 4 3 2 1

We can skip that by doing `assert_script_run qq(btrfs balance start $dev_path --full-balance), timeout => 600;`

Reply:
addressed on 8d47f41